File: code_snippets/java/src/main/java/usecases/microservices/ProductService.java
package usecases.microservices;

import dev.restate.sdk.ObjectContext;
import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.VirtualObject;

@VirtualObject
public class ProductService {

  @Handler
  public boolean reserve(ObjectContext ctx) {
    return true;
  }
}

File: code_snippets/kotlin/src/main/kotlin/usecases/microservices/ProductService.kt
package usecases.microservices

import dev.restate.sdk.annotation.Handler
import dev.restate.sdk.annotation.VirtualObject
import dev.restate.sdk.kotlin.ObjectContext

@VirtualObject
class ProductService {

    @Handler
    suspend fun reserve(ctx: ObjectContext): Boolean {
        return true;
    }
}